tools/roadmap/validate_dag.py

```
@@ -0,0 +1,69 @@
+    if not MMD.exists():
+        print("validate_dag: missing docs/ROADMAP-DAG.mmd", file=sys.stderr)
+        return 1
+    text = MMD.read_text(encoding='utf-8')
+    nodes, edges_hard, _ = parse_mmd(text)
+    # Map node id -> issue number
+    node_to_issue = {n: num for (n,num) in nodes}
+
+    # Basic syntax lint via mermaid-cli if available (optional)
+    try:
+        run(["docker", "run", "--rm", "-v", f"{MMD.parent.resolve() }:/data", "minlag/mermaid-cli", "-i", "/data/ROADMAP-DAG.mmd", "-o", "/data/.lint.svg"])
+    except Exception as e:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)